drivers/gnss/gnss_u_blox_m10.c

```diff
@@ -0,0 +1,940 @@
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = sizeof(data->chat_receive_buf),
+		.delimiter = ubx_m10_char_delimiter,
+		.delimiter_size = ARRAY_SIZE(ubx_m10_char_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+	};
+
+	return modem_chat_init(&data->chat, &chat_config);
+}
+
+static int ubx_m10_init_ubx(const struct device *dev)
+{
+	struct ubx_m10_data *data = dev->data;
+
+	const struct modem_ubx_config ubx_config = {
+		.user_data = data,
+		.receive_buf = data->ubx_receive_buf,
+		.receive_buf_size = sizeof(data->ubx_receive_buf),
+		.work_buf = data->ubx_work_buf,
+		.work_buf_size = sizeof(data->ubx_work_buf),
+	};
+
+	return modem_ubx_init(&data->ubx, &ubx_config);
+}
+
+static int ubx_m10_modem_module_switch(const struct device *dev, enum MODEM_MODULE release,
+				       enum MODEM_MODULE attach)
+{
+	struct ubx_m10_data *data = dev->data;
+	int ret;
+
+	switch (release) {
+	case MODEM_MODULE_CHAT:
+		modem_chat_release(&data->chat);
+		break;
+	case MODEM_MODULE_UBX:
+		modem_ubx_release(&data->ubx);
+		break;
+	default:
+		return -1;
+	}
+
+	switch (attach) {
+	case MODEM_MODULE_CHAT:
+		ret = modem_chat_attach(&data->chat, data->uart_pipe);
+		break;
+	case MODEM_MODULE_UBX:
+		ret = modem_ubx_attach(&data->ubx, data->uart_pipe);
+		break;
+	default:
+		return -1;
+	}
+
+	if (ret < 0) {
+		modem_pipe_close(data->uart_pipe);
+	}
+
+	return ret;
+}
+
+static int ubx_m10_modem_ubx_run_script(const struct device *dev,
+					struct modem_ubx_script *modem_ubx_script_tx)
+{
+	struct ubx_m10_data *data = dev->data;
+	int ret;
+
+	ret = ubx_m10_modem_module_switch(dev, MODEM_MODULE_CHAT, MODEM_MODULE_UBX);
+	if (ret < 0) {
+		goto reset_modem_module;
+	}
+
+	ret = modem_ubx_run_script(&data->ubx, modem_ubx_script_tx);
+	if (ret < 0) {
+		goto reset_modem_module;
+	}
+
+reset_modem_module:
+	ret |= ubx_m10_modem_module_switch(dev, MODEM_MODULE_UBX, MODEM_MODULE_CHAT);
+
+	return ret;
+}
+
+static void ubx_m10_modem_ubx_script_fill(const struct device *dev)
+{
+	struct ubx_m10_data *data = dev->data;
+
+	data->script.request = (struct ubx_frame_t *) data->request_buf;
+	data->script.response = (struct ubx_frame_t *) data->response_buf;
+	data->script.match = (struct ubx_frame_t *) data->match_buf;
+	data->script.retry_count = RETRY_DEFAULT;
+	data->script.timeout = K_MSEC(MODEM_UBX_SCRIPT_TIMEOUT_MS);
+}
+
+static int ubx_m10_modem_ubx_script_init(const struct device *dev, void *payload, uint16_t payld_sz,
+					 enum ubx_msg_class msg_cls, enum ubx_config_message msg_id)
+{
+	int ret;
+	struct ubx_m10_data *data = dev->data;
+	struct ubx_cfg_ack_data match_payload = {
+		.message_class = msg_cls,
+		.message_id = msg_id,
+	};
+
+	(void) ubx_m10_modem_ubx_script_fill(dev);
```

Comment:
```suggestion
	(void)ubx_m10_modem_ubx_script_fill(dev);
```

same for the rest of the file